### PR TITLE
feat(toolbox): axe the old toolbox

### PR DESCRIPTION
**Pull request by saghul**
This PR takes The Bulldozer Approach (R): removes the old toolbox and lots of
associated code, though not all of it. Subsequent cleanups will follow.

### CODE REVIEW

**Comment by saghul** [comment]
Things not covered by this PR:

* CSS cleanup
* Recording.js cleanup - I reckon we no longer need most of it
* Full-screen detection: it should be moved to a middleware instead of having conference tell us the result of our action
* Popovers for buttons are no longer a thing, gut out any remaining code
* The DTMF button is now dead, kill associated code or fix it.

**Comment by saghul** [comment]
Jenkins please test.

File: react/features/toolbox/actions.web.js [hunk 1]
```diff
@@ -1,73 +1,24 @@
...
- */
-export function checkAutoEnableDesktopSharing(): Function {
-    return () => {
-        // XXX Should use dispatcher to toggle screensharing but screensharing
-        // hasn't been React-ified yet.
-        if (isButtonEnabled('desktop')
-                && config.autoEnableDesktopSharing) {
-            APP.UI.eventEmitter.emit(UIEvents.TOGGLE_SCREENSHARING);
-        }
-    };
```
Comment (virtuacoplenny) [inline react/features/toolbox/actions.web.js]: This is a funny one. Isn't is the same as config.startScreenSharing?

File: react/features/toolbox/actions.native.js [hunk 2]
```diff
@@ -203,51 +159,3 @@ export function setToolboxVisible(visible: boolean): Object {
...
- */
-export function toggleToolbarButton(buttonName: string): Function {
-    return (dispatch: Dispatch, getState: Function) => {
-        const button = getButton(buttonName, getState());
-
-        dispatch(setToolbarButton(buttonName, {
-            toggled: !button.toggled
-        }));
-    };
-}
```
Comment (virtuacoplenny) [inline react/features/toolbox/actions.native.js]: I get why these functions existed, to bridge a gap between the react-toolbar and the non-reduxified code, but I'm happy to see them go.

**Comment by virtuacoplenny** [comment]
Someone had fun...

Looks good to me. I'll do follow up clean up after this gets merged and I get time. I'll fix the conflicts and approve but I'll have to wait on confirmation that this can get merged.

File: modules/UI/UI.js [hunk 3]
```diff
@@ -298,8 +291,6 @@ UI.initConference = function() {
         UI.setUserEmail(id, email);
     }
 
-    dispatch(checkAutoEnableDesktopSharing());
```
[the remark this answers is not in the document]
Comment (damencho) [reply modules/UI/UI.js]: @virtuacoplenny autoEnableDesktopSharing was introduced to be able to do some desktop sharing tests, which we no longer do. I also think we had removed that test from torture. And now we have the config we can pass startScreenSharing, so I think we can remove that.

[the remark this answers is not in the document]
Comment (virtuacoplenny) [reply modules/UI/UI.js]: I'll remove autoEnableDesktopSharing from config.js during a follow up clean up commit.

[the remark this answers is not in the document]
Comment (paweldomas) [reply modules/UI/UI.js]: hmm maybe this was used in torture ?

**Comment by saghul** [comment]
> I have to leave notice on the mailing list that the functionality of specifying toolbar button location has been removed.

That will go back in with the new changes I'm making.

**Comment by virtuacoplenny** [comment]
An email was sent out a couple days ago to the jitsi community stating customization of toolbar button placement will be removed. No objections have been received so far. I'll wait until Monday morning, get another confirmation, and merge.

**Pull request merged**